### PR TITLE
front: add a check to display suggestions for operational points

### LINKED ISSUE #7178: Rocket pathfinding suggestions remains after selecting a PR
### What happened?

I wrote Lyon in the rocket pathfinding input order to go from Macon to Lyon, and after clicking on Lyon-Part-Dieu the suggestions remains, even though i already chose the PR i wanted:
image

### What did you expect to happen?

I'm expecting the suggestions to disappear after selecting the one i wanted.

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to any pathfinding input (STDCM or train creation)
2. Search for "Lyon"
3. Click on Lyon-Part-Dieu
4. Notice the bug

### What operating system, browser and environment are you using?

- **Browser**: Chrome
- **OS**: W11
- **Env**: staging internal

### OSRD version (top right corner `Account` button > `Informations`)

8f43d76

### PR DESCRIPTION
**Pull request by theocrsb**
closes #7178

### CODE REVIEW

File: front/src/modules/pathfinding/components/Pathfinding/TypeAndPathV2.tsx [hunk 1]
```diff
@@ -223,6 +223,13 @@ const TypeAndPathV2 = () => {
     setInitialCursorPositionRem(0);
   }, []);
 
+  const isSortedSearchResultsDisplayed = () => {
+    const trigrams = debouncedInputText.split(' ');
+    const opListFiltered = opList.filter((op) => op.name !== undefined);
+    if (trigrams.length === opListFiltered.length) return false;
+    return true;
```
Comment (SarahBellaha) [inline front/src/modules/pathfinding/components/Pathfinding/TypeAndPathV2.tsx]:
```suggestion
    return trigrams.length !== opListFiltered.length;
```

Comment (theocrsb) [reply front/src/modules/pathfinding/components/Pathfinding/TypeAndPathV2.tsx]: Done :)

**Review by SarahBellaha — APPROVED** [review]
LGTM :)

File: front/src/modules/pathfinding/components/Pathfinding/TypeAndPathV2.tsx [hunk 2]
```diff
@@ -223,6 +223,12 @@ const TypeAndPathV2 = () => {
     setInitialCursorPositionRem(0);
   }, []);
 
+  const isSortedSearchResultsDisplayed = () => {
```
Comment (clarani) [inline front/src/modules/pathfinding/components/Pathfinding/TypeAndPathV2.tsx]: I would rather use a useMemo here so `isSortedSearchResultsDisplayed` is a const and not a function, can you change that ? :)

The dependencies would be `[debouncedInputText, opList]`

**Review by clarani — COMMENTED** [review]
LGTM & tested, feel free to resolve the comment once you are done :white_check_mark:

File: front/src/modules/pathfinding/components/Pathfinding/TypeAndPathV2.tsx (hunk 2 above)
Comment (theocrsb) [reply front/src/modules/pathfinding/components/Pathfinding/TypeAndPathV2.tsx]: Done :)

**Review by clarani — COMMENTED** [review]
LGTM :white_check_mark:

**Review by clarani — APPROVED** [review]

**Pull request merged**